src/main/java/mate/academy/dao/impl/MovieSessionDaoImpl.java

```diff
@@ -0,0 +1,79 @@
+package mate.academy.dao.impl;
+
+import java.time.LocalDate;
+import java.time.LocalTime;
+import java.util.List;
+import java.util.Optional;
+import mate.academy.dao.MovieSessionDao;
+import mate.academy.exception.DataProcessingException;
+import mate.academy.lib.Dao;
+import mate.academy.model.MovieSession;
+import mate.academy.util.HibernateUtil;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.Transaction;
+import org.hibernate.query.Query;
+
+@Dao
+public class MovieSessionDaoImpl implements MovieSessionDao {
+    private final SessionFactory factory = HibernateUtil.getSessionFactory();
+
+    @Override
+    public MovieSession add(MovieSession movieSession) {
+        Transaction transaction = null;
+        Session session = null;
+        try {
+            session = factory.openSession();
+            transaction = session.beginTransaction();
+            session.save(movieSession);
+            transaction.commit();
+            return movieSession;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("Can't insert movieSession " + movieSession, e);
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
+    }
+
+    @Override
+    public Optional<MovieSession> get(Long id) {
+        try (Session session = factory.openSession()) {
+            Query<MovieSession> getMovieSessionById = session
+                    .createQuery("FROM MovieSession ms"
+                            + " left join fetch ms.movie"
+                            + " left join fetch ms.cinemaHall"
+                            + " WHERE ms.id = :id", MovieSession.class);
+            getMovieSessionById.setParameter("id", id);
+            return Optional.ofNullable(session.get(MovieSession.class, id));
+        } catch (Exception e) {
+            throw new DataProcessingException("Can't get a MovieSession by id: " + id, e);
+        }
+    }
+
+    @Override
+    public List<MovieSession> findAvailableSessions(Long movieId, LocalDate date) {
+        try (Session session = factory.openSession()) {
+            Query<MovieSession> findAvailableSessions = session
+                    .createQuery("FROM MovieSession ms"
+                            + " left join fetch ms.movie"
+                            + " left join fetch ms.cinemaHall"
```

Comment:
why did you put fetch type LAZY on fields movie and cinemaHall and have to use left join fetch now?